pandas/core/groupby/groupby.py

```diff
@@ -1731,8 +1732,30 @@ def count(self):
         Series or DataFrame
             Count of values within each group.
         """
-        # defined here for API doc
-        raise NotImplementedError
+        data = self._get_data_to_aggregate()
+        ids, _, ngroups = self.grouper.group_info
+        mask = ids != -1
+
+        def hfunc(bvalues: ArrayLike) -> ArrayLike:
```

Comment:
this looking like a pretty generic function and could be factored out (followon)